src/gt4py/next/program_processors/codegens/gtfn/gtfn_ir_to_gtfn_im_ir.py

```diff
@@ -330,6 +330,19 @@ def visit_FunctionDefinition(
             id=node.id, params=node.params, fun=[*self.imp_list_ir, ReturnStmt(ret=ret)]
         )
 
+    def visit(self, node: concepts.RootNode, **kwargs: Any) -> Any:
+        kwargs = {
+            **kwargs,
+            "inside_fun": kwargs.get("inside_fun", False)
+            or isinstance(node, gtfn_ir.FunctionDefinition),
+        }
+
+        # only transform things inside of function definitions and leave the rest as is
+        if kwargs["inside_fun"]:
+            return super().visit(node, **kwargs)
+        else:
+            return self.generic_visit(node, **kwargs)
```

Comment:
Please add comments what the 2 cases do. It's a non standard pattern and not obvious to me what's called here.